File: app/src/main/java/soup/movie/ui/main/MainContract.java
package soup.movie.ui.main;

import android.support.annotation.IntDef;
import android.support.annotation.NonNull;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

import soup.movie.ui.BaseContract;

public interface MainContract {

    @Retention(RetentionPolicy.SOURCE)
    @IntDef({TAB_MODE_NOW, TAB_MODE_PLAN, TAB_MODE_SETTINGS})
    @interface TabMode {}

    int TAB_MODE_NOW = 10;
    int TAB_MODE_PLAN = 20;
    int TAB_MODE_SETTINGS = 30;

    interface Presenter extends BaseContract.Presenter<View> {
        void setTabMode(@TabMode int mode);
    }

    interface View extends BaseContract.View {
        void render(@NonNull MainViewState viewState);
    }
}

File: app/src/main/java/soup/movie/ui/main/MainContract.kt
package soup.movie.ui.main

import soup.movie.data.model.MovieId
import soup.movie.settings.impl.LastMainTabSetting
import soup.movie.ui.LegacyBaseContract

interface MainContract {

    interface Presenter : LegacyBaseContract.Presenter<View> {

        fun setCurrentTab(mode: LastMainTabSetting.Tab)

        fun requestMovie(movieId: MovieId?)
    }

    interface View : LegacyBaseContract.View {

        fun render(viewState: MainViewState)

        fun execute(action: MainActionState)
    }
}
